Classification

fosterlynn opened the pull request:
I'm working on clarifying the naming and concepts in the abstract/recipe layer (which I also renamed in the picture).  It has become more clear (from some experience with FairCoop and Mutual Aid Network) that we will want to support many kinds of data structures for people to classify their resources, processes, etc.  The term "taxonomy item" is limiting, because to people who work with taxonomies, it means "part of a tree", and most other people don't know what it is, so it is a bit intimidating.  People have lots of opinions and about how to classify things, and at this moment I'm looking for a neutral word that can fit into however people want to do it, or just do do it already.

Bob suggested "classification", which seems pretty good to me.  If anyone has a better general term, please suggest it.

I didn't fill in any of the recipe structure yet, just some of the base classifications.

fosterlynn commented:
Hmmm, I don't know why the scope commit is here.  I hope I don't have things too mixed up in git land.  I did forget to initially branch, so I branched after I added my stuff (making my local master in need of cleanup) but that doesn't explain the scope stuff.

fosterlynn commented:
I should also explicitly note that I removed the Transformation and Transportation subclasses of Process, because although useful, I think that is better dealt with in the "classification" layer.

Also, I think some people might want to include both some transformation and some transportation in one process, so don't see a need to limit it.

bhaugen commented:
This might be useful:
https://board.net/p/Classification
(or not)

elf-pavlik commented:
Besides renaming this PR adds
* ProcessClasification
* ExchangeClasification

I think we should have some examples of them first. I also wonder if we can find references for those examples in wikidata eg.

* transport - https://www.wikidata.org/wiki/Q7590
* packaging activity - https://www.wikidata.org/wiki/Q29585952

or

* barter - https://www.wikidata.org/wiki/Q484830
* trade - https://www.wikidata.org/wiki/Q601401
* gift - https://www.wikidata.org/wiki/Q184303

in Vientos we plan to use those 3 above + 'time exchange'

I would also like to make sure that when we talk about Classification/Category, they only have
* `@id` (IRI)
* `skos:label` - preferably in number of languages
* `subOf` & `suberOf` - for multi parent hierarchies

Besides those 3 we will never try to dump any more information into categories and only keep them as taxonomy references. For now we can mark as TBD how to deal with reverse/inverse relationships like we have for agents...

fosterlynn commented:
@elf-pavlik thanks for chiming in!

>I think we should have some examples of them first.

Good idea.  And I like all your examples.  Should I start up a page for the gitbook to describe them?  It will be much more developed when we figure out how to do process/exchange recipes I think.  But could be started up.

>I would also like to make sure that when we talk about Classification/Category, they only have
>    @id (IRI)
>    skos:label - preferably in number of languages
>    subOf & suberOf - for multi parent hierarchies

Agree, I will change the properties to @id and skos:label.  (I guess @id will be automatic/assumed.)  I disagree with modeling the hierarchies in VF though, after a lot of thought stemming from recent experiences.  Basically, not every kind of classification will be a taxonomy tree.  If we model the tree structure, we also have to model other structures people will use.  

My opinion at this moment is that VF should stay out of that business, it has a lot of complexities and opinions, I am willing to bet it has already been done, and it is outside our core purpose.  If we just have the single references, a person can go to for example https://www.wikidata.org/wiki/Q484830 and find the tree structure from there.  Or whatever classification structure any one item has attached to it.

Re. agent relationship classification, maybe I should just change it back to AgentRelationshipRole?  I don't think it will be part of recipes.  And it is different since it is a relationship not a thing.  Or just keep it as AgentRelationshipClassification with the extra property?  I'm totally fine with either.

fosterlynn commented:
@elf-pavlik should we use skos:label or rdfs:label?  Or vf:label which has range of skos:label or rdfs:label?  Or something else?

Hmmmm, here is what shows up in the @context in the current version of the vocabulator, which we also need to fix:

```
        "label": "skos:prefLabel",
        "labelMap": {
            "@container": "@language",
            "@id": "skos:prefLabel"
        },
```
and
```
        "rdfs:label": {
            "@container": "@language"
        },
```

Please help with syntax!  Thanks!

fosterlynn commented:
To clarify also, the thing that has kicked me into gear to try to nail down some of these things is a mapping effort we have started with @oceatoon and others working on Communecter, to map VF into a vocabulary translator they have developed.  And there is another one in the works to map VF to the Assemblee Virtuelle vocabulary.  I hope we can influence both of these, since they don't have all the constructs we need.  And related to those, the Open App Ecosystem is showing signs of life. :smile:   https://www.loomio.org/g/exAKrBUp/open-app-ecosystem

bhaugen commented:
Another part of that vocabulary translation would be to translate back and forth between VF and the Data Food Consortium where we have had some conversations. @djodjoni was part of them, too.

oceatoon commented:
does DFC have their ontology mapped out ?
it would be fairly easy if there's an API out there ?

On Mon, Aug 21, 2017 at 8:31 PM, Bob Haugen <notifications@github.com>
wrote:

> Another part of that vocabulary translation would be to translate back and
> forth between VF and the Data Food Consortium where we have had some
> conversations. @djodjoni <https://github.com/djodjoni> was part of them,
> too.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/valueflows/valueflows/pull/236#issuecomment-323792425>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AALuTBRY4wCPinq-5yRsaAj8G5uQEaBfks5sabDRgaJpZM4O9dwH>
> .
>



-- 
Tibor Katelbach – Senior Developer - Gérant Bénévole Association Open Atlas



projet : https://www.communecter.org/
..·><((((º>·´¯`·.¸¸..><((((º>.¸¸..·.¸¸..><((((º>.¸¸..
La Riviere, La Reunion
e-mail : oceatoon@gmail.com (toujours sur GTalk) <oceatoon@gmail.com>
skype : oceatoon
Fixe : +262+262343686
Mobile : +262-693918532

Save a tree...please don't print this e-mail unless you really need to.

elf-pavlik commented:
I would start with probably the most commonly used `rdfs:label` and then adjust it to something like `skos:prefLabel` once we have a clear reason to do it.

Should we have one parent and three subclasses
* Classification
  * ResourceClassification
  * ProcessClassification
  * ExchangeClassification

Something we should keep in mind, in practice when we use references from common knowledge sources like Wikidata, they will not include statements like `wd:Q7590 rdf:type vf:ProcessClasification` so in practice software wouldn't rely on those referenced entities having specific classes but instead use some other construct to group those references by those which apply to resource, process or exchange. For example some kind of list/index of choices for each resource, process and exchange. Preferably we should define requirements first based on some use cases and then try to propose particular modeling approaches.

bhaugen commented:
@elf-pavlik 
> in Vientos we plan to use those 3 above + 'time exchange'

We should take a look at Vientos again. Are those implemented? Code published? Would be instructive to learn how you are handling those concepts and others, and then update stuff here in VF.

bhaugen commented:
@oceatoon 
> does DFC have their ontology mapped out ?

We were talking to @myriamboure who was going offline for a couple of months, I think July and August. Maybe that mention will alert her if she has returned? But I doubt it. September!

fosterlynn commented:
>Something we should keep in mind, in practice when we use references from common knowledge sources like Wikidata, they will not include statements like wd:Q7590 rdf:type vf:ProcessClasification so in practice software wouldn't rely on those referenced entities having specific classes but instead use some other construct to group those references by those which apply to resource, process or exchange.

I defer to others for if we have a superclass of the classifications, although I don't see the use yet for it.  Seems to me that people in any case will have to use existing search mechanisms in various sources to find the classification that works for them, we are not going to index any existing taxonomies.  But for our purposes for constructing recipes, we will want to know the difference, so once an existing @id is associated to a vf:XXClassification, which is up to the user, that is what is needed.  Make sense or am I missing something?  Or was this even clear?

>Preferably we should define requirements first based on some use cases and then try to propose particular modeling approaches.

The only use cases I am very familiar with are ones where they have defined their own taxonomies or non-taxonomies (flat category lists, resource types with faceted classification).  In those cases, they already know if it is for resource, process, or exchange.  Is there a Vientos use case that would be useful?

I propose we start without the superclass (we can add it if we want to later), keep the class differentiations we have, as a pretty sure starting point, as long as we can agree on names.

fosterlynn commented:
The last commit changed properties per discussion, changed AgentRelationshipRole name back.  Also added scoping document to index, seems it was forgotten before.

fosterlynn commented:
Last commit added some explanation.  Please feel to add to this, correct, refine, etc.

fosterlynn commented:
Almost forgot: I also changed vf:Action in the UML to more closely map to the rdf based structure.

oceatoon commented:
@elf-pavlik I really have to show you what we are doing for Classifications and Wikidata + Wikipedia 
also very linked to the translation engine , 
I presented it to Lynn and Bob here 
https://www.youtube.com/watch?v=9Yb5bomm-QE 
the interesting wikipedia part starts at 50min

elf-pavlik commented:
What do you think about *fooClassifiedAs* instead of *fooClassifiedBy* when naming properties?
or *fooClassifiedUnder*

fosterlynn commented:
>What do you think about fooClassifiedAs instead of fooClassifiedBy when naming properties?
or fooClassifiedUnder

I like fooClassifiedAs

fosterlynn commented:
>I like fooClassifiedAs

done

gcassel commented:
>Bob suggested "classification", which seems pretty good to me. If anyone has a better general term, please suggest it.

FWIW, I currently use *[type](https://github.com/gcassel/Modular-Organization-Terminology/blob/master/terms/type.md)* and *typology* instead of *class* and *classification* as fundamental/root terms in [Modular Organization Terminology](https://github.com/gcassel/Modular-Organization-Terminology).     

I intentionally avoided the term *class* in early MOT development because of the often emotionally charged concept of *social class*.   (Like, lords and ladies in contrast to peasants-- that sort of thing.)  Social class is a derivative meaning, of course, but it's referred to simply as "class" in a great deal of culturally critical media.  That tendency can create confusion if "class" is used more generically.

For example:  I often refer (*critically*) to the idea of "types of people" in online discussions.  I definitely would *not* feel comfortable using the term "classes of people" instead, because IMO it'd confuse and mislead random viewers.

I believe that that specific, occasional problem with the use of 'class' in natural language could indirectly complicate efforts to create highly legible data structures.   

However, I don't have a decided personal preference on the subject-- and, I may be failing to recognize some potential problems with "type".    (So, I'd appreciate any critical feedback.)

bhaugen commented:
This, of course, is hilarious. We started out (in REA and NRP) with Type. In VF, some people ( @elf-pavlik ?) did not like Type, which has some other meanings in the Semantic Web context. So we foofulled around looking for other acceptable euphemisms, and apparently I suggested "Classification".

And of course, as is often the case with these word circles, @gcassel wants to bring us back to Type.

Almost all words in this semantic category are highly overloaded: that is, used in different domains and contexts with different meanings. I don't know a good solution to this predicament. Yet. 

I think it might involve all of you appointing me as Dictator. Good luck with that...I decline...

gcassel commented:
I really don't have a decided preference on the question of class *or* type.    I stated a specific concern about *class*, which I've possibly mentioned in VF before-- but if I ever did that, I certainly didn't do it  clearly, concisely or promptly.   (Sorry about that.)

In the broader scheme of shared languages, @bhaugen , I doubt that such either/or decisions will be important.   (And I'm guessing you might agree with me?)    In theory, a whole bunch of communities could decide that class and type are *practically pure synonyms* which are interchangeable in some, most or perhaps (with deeply resilient structures) *all* formal contexts.    

In fact, I suppose it'd be possible right now for VF to define both terms as being officially interchangeable-- but, I also suppose that that'd seem undesirable for a relatively small and strictly focused formal vocabulary?

bhaugen commented:
I think we need to support alternate names for all concepts. And we are also talking to people about translation from one vocabulary to another.
